### PR TITLE
Enhancement: Redirect to /contribute when /donate is disabled

### PR DESCRIPTION
**Pull request by saideepesh000**
Resolve [https://github.com/opencollective/opencollective/issues/4066](https://github.com/opencollective/opencollective/issues/4066)

# Description
This PR redirects users to /collective when /donate is disabled

# Gif
redirect

### CODE REVIEW

File: components/contribution-flow/ContributionBlocker.js [hunk 1]
```diff
@@ -49,6 +50,16 @@ const msg = defineMessages({
...
  * From received params, see if there's anything preventing the contribution
  */
+const Redirect = ({ to }) => {
+  const router = useRouter();
+
+  useEffect(() => {
+    router.push(to);
+  }, [to]);
+
+  return null;
```
Comment (Betree) [inline components/contribution-flow/ContributionBlocker.js:60]: It would be nice to show a loading spinner while the user is being redirected so that the page doesn't look empty

```suggestion
  return <Loading />;
```

File: components/contribution-flow/ContributionBlocker.js [hunk 2]
```diff
@@ -83,15 +94,18 @@ export const getContributionBlocker = (loggedInUser, account, tier, shouldHaveTi
     return { reason: CONTRIBUTION_BLOCKER.TIER_MISSING, type: 'warning', showOtherWaysToContribute: true };
   } else if (tier && isTierExpired(tier)) {
     return { reason: CONTRIBUTION_BLOCKER.TIER_EXPIRED, type: 'warning', showOtherWaysToContribute: true };
-  } else if (account.settings.disableCustomContributions && !tier) {
```
Comment (Betree) [inline components/contribution-flow/ContributionBlocker.js:86]: I don't get how this works, since `ContributionBlocker` will not be rendered if `getContributionBlocker` returns `null`

**Review by Betree — COMMENTED** [review]
We also need to cover `pages/contribution-flow.js`.

Also, could you share a screencast of the feature so that we can easily preview how it looks like?

**Comment by saideepesh000** [comment]
Sure, will update soon 👍

File: pages/contribution-flow.js [hunk 3]
```diff
@@ -132,6 +135,9 @@ class NewContributionFlowPage extends React.Component {
     }
 
     const contributionBLocker = getContributionBlocker(LoggedInUser, account, tier, Boolean(this.props.tierId));
+    if (contributionBLocker.reason === 'NO_CUSTOM_CONTRIBUTION') {
```
Comment (SudharakaP) [inline pages/contribution-flow.js:138]: It's better to import the constant and use it rather than hard-coding so that if it changes in the future we can just change it at a single location. 😄 

```suggestion
    if (contributionBLocker.reason === CONTRIBUTION_BLOCKER.NO_CUSTOM_CONTRIBUTION) {
```

File: pages/embed/contribution-flow.js [hunk 4]
```diff
@@ -149,8 +152,11 @@ class NewContributionFlowPage extends React.Component {
     }
 
     const contributionBLocker = getContributionBlocker(LoggedInUser, account, tier, Boolean(this.props.tierId));
+    if (contributionBLocker.reason === 'NO_CUSTOM_CONTRIBUTION') {
```
Comment (SudharakaP) [inline pages/embed/contribution-flow.js:155]:
```suggestion
    if (contributionBLocker.reason === CONTRIBUTION_BLOCKER.NO_CUSTOM_CONTRIBUTION) {
```

File: pages/contribution-flow.js (hunk 3 above)
Comment (saideepesh000) [reply pages/contribution-flow.js:138]: Oh.. 🧠 sure, any other changes should I make...

**Comment by saideepesh000** [comment]
@Betree  Should I create a separate `Redirect` component in `components` folder or should I directly add proptypes for `Redirect` in `ContributionBlocker` .

File: components/contribution-flow/ContributionBlocker.js [hunk 5]
```diff
@@ -49,6 +51,16 @@ const msg = defineMessages({
 /**
  * From received params, see if there's anything preventing the contribution
  */
+export const Redirect = ({ to }) => {
```
Comment (SudharakaP) [inline components/contribution-flow/ContributionBlocker.js:54]: I think this `to` needs to be added to `Proptypes`. The linter complains about it as well. 😃

**Review by SudharakaP — CHANGES_REQUESTED** [review]

**Comment by SudharakaP** [comment]
> @Betree @SudharakaP Should I create a separate `Redirect` component in `components` folder or should I directly add proptypes for `Redirect` in `ContributionBlocker` .

@saideepesh000 : My thinking is since `Redirect` only acts as a wrapper around `Loading` and only used when `ContributionBlocker` is imported (using `getContributionBlocker`), it's okay to keep it within as you've done. Let us know if you think otherwise @Betree 😄

**Comment by saideepesh000** [comment]
Anything I need to do ?, may be it's time start writing tests as well :)

File: pages/contribution-flow.js [hunk 6]
```diff
@@ -134,6 +138,9 @@ class NewContributionFlowPage extends React.Component {
     }
 
     const contributionBLocker = getContributionBlocker(LoggedInUser, account, tier, Boolean(this.props.tierId));
+    if (contributionBLocker.reason === CONTRIBUTION_BLOCKER.NO_CUSTOM_CONTRIBUTION) {
```
Comment (SudharakaP) [inline pages/contribution-flow.js:141]: @saideepesh000 : I don't see this `NO_CUSTOM_CONTRIBUTION` being defined anywhere. Hence I think this should fail. This is probably why the E2E tests are failing. 🤔

**Review by SudharakaP — CHANGES_REQUESTED** [review]

File: pages/contribution-flow.js (hunk 6 above)
Comment (saideepesh000) [reply pages/contribution-flow.js:141]: I actually imported `CONTRIBUTION_BLOCKER` from [ContributionBlocker.js](https://github.com/opencollective/opencollective-frontend/blob/3f25bbbcb324d58747927a6f5a450b6206bc5f1b/components/contribution-flow/ContributionBlocker.js#L12)

Comment (SudharakaP) [reply pages/contribution-flow.js:141]: @saideepesh000 : Ops my mistake. But still the E2E tests are failing since the `/donate` url seems to error out in the server. 

https://opencollective-frontend-b1gll8btb-opencollective.vercel.app/apex/donate

File: components/contribution-flow/ContributionBlocker.js (hunk 5 above)
Comment (Betree) [inline components/contribution-flow/ContributionBlocker.js:54]: This component is not really related to `ContributionBlocker`, so I think it deserves its own file (ie. `.components/Redirect.js`)

File: pages/embed/contribution-flow.js [hunk 7]
```diff
@@ -149,6 +153,9 @@ class NewContributionFlowPage extends React.Component {
     }
 
     const contributionBLocker = getContributionBlocker(LoggedInUser, account, tier, Boolean(this.props.tierId));
+    if (contributionBLocker.reason === CONTRIBUTION_BLOCKER.NO_CUSTOM_CONTRIBUTION) {
+      return <Redirect to={`${account.slug}/contribute`} />;
+    }
```
Comment (Betree) [inline pages/embed/contribution-flow.js:158]: This will redirect to a page that is not embed-friendly. We should rather keep the error message in there.

```suggestion
```

Comment (Betree) [inline pages/embed/contribution-flow.js:158]: This will crash if there's no contribution blocker. We should move this inside the `if (contributionBLocker) {`

File: pages/embed/contribution-flow.js [hunk 8]
```diff
@@ -149,6 +153,9 @@ class NewContributionFlowPage extends React.Component {
     }
 
     const contributionBLocker = getContributionBlocker(LoggedInUser, account, tier, Boolean(this.props.tierId));
```
Comment (SudharakaP) [inline pages/embed/contribution-flow.js:155]: Minor typo in the naming; 

```suggestion
    const contributionBlocker = getContributionBlocker(LoggedInUser, account, tier, Boolean(this.props.tierId));
```

Comment (saideepesh000) [reply pages/embed/contribution-flow.js:155]: Hey @SudharakaP,  I think it was done purposefully to differentiate it from `ContributionBlocker` component

Comment (SudharakaP) [reply pages/embed/contribution-flow.js]: @saideepesh000 : I don't think so. `ContributionBlocker` starts with uppercase `C` anyways. This seems more like a typo. 😄

File: components/contribution-flow/ContributionBlocker.js [hunk 9]
```diff
@@ -49,6 +49,7 @@ const msg = defineMessages({
 /**
  * From received params, see if there's anything preventing the contribution
  */
+
```
Comment (Betree) [inline components/contribution-flow/ContributionBlocker.js:52]: Let's keep the function and its documentation together

```suggestion
```

**Review by Betree — COMMENTED** [review]
That works great! Let's address https://github.com/opencollective/opencollective-frontend/pull/6046#discussion_r605730202 and we should be good to go :rocket:

**Comment by saideepesh000** [comment]
Any help in squashing, I don not know how to do it😢

**Comment by saideepesh000** [comment]
Lol, I messed up something, going to delete this PR and open a new one. 😢 😭

**Comment by Betree** [comment]
Moved to https://github.com/opencollective/opencollective-frontend/pull/6156
